src/vs/platform/quickinput/browser/quickInput.ts

```diff
@@ -530,6 +530,7 @@ export class QuickPick<T extends IQuickPickItem> extends QuickInput implements I
 	private _ok: boolean | 'default' = 'default';
 	private _customButton = false;
 	private _customButtonLabel: string | undefined;
+	private _customButtonIcon: ThemeIcon | undefined;
```

Comment:
I'd rather you keep using label, but have it be `$(icon-name)`

Reply:
I think that the button ends up having a little more padding on each side, but I think that's fine.
<img width="566" alt="image" src="https://github.com/microsoft/vscode/assets/31675041/c4a3e80d-87ca-42b2-9c13-90194ed05389">